pocs/filterwheel/libefw.py

```diff
@@ -0,0 +1,181 @@
+import ctypes
+import enum
+
+from pocs.camera.sdk import AbstractSDKDriver
+from pocs.utils import error
+from pocs.utils.library import load_library
+
+
+class EFWDriver(AbstractSDKDriver):
+    # Because ZWO EFW library isn't linked properly have to manually load libudev
+    # in global mode first, otherwise get undefined symbol errors.
+    _libudev = load_library('udev', mode=ctypes.RTLD_GLOBAL)
+
+    def __init__(self, library_path=None, **kwargs):
+        """Main class representing the ZWO EFW library interface.
+
+        On construction loads the shared object/dynamically linked version of the EFW SDK library,
+        which must be already installed (see https://astronomy-imaging-camera.com/software-drivers).
+
+        The name and location of the shared library can be manually specified with the library_path
+        argument, otherwise the ctypes.util.find_library function will be used to try to locate it.
+
+        Args:
+            library_path (str, optional): path to the libary e.g. '/usr/local/lib/libEFWFilter.so'
+
+        Returns:
+            `~pocs.filter.libefw.EFWDriver`
+
+        Raises:
+            pocs.utils.error.NotFound: raised if library_path not given & find_library fails to
+                locate the library.
+            OSError: raises if the ctypes.CDLL loader cannot load the library.
+        """
+        super().__init__(name='EFWFilter', library_path=library_path, **kwargs)
+
+    # Methods
+
+    def get_SDK_version(self):
+        """Get the version for the SDK."""
+        # EFW SDK has no way to access SDK version.
+        return "Unknown"
+
+    def get_devices(self):
+        """Get connected device UIDs and corresponding device nodes/handles/IDs.
+        """
+        # EFW SDK has no way to access any unique identifier for connected filterwheels.
+        # Construct an ID (probably not deterministic, in general) from combination of
+        # product code and filterwheel ID.
+        n_filterwheels = self.get_num()  # Nothing works if you don't call this first.
+        products = self.get_product_ids()  # Will raise error.NotFound if no filterwheels
+        ids = [self.get_ID(i) for i in range(n_filterwheels)]
+        filterwheels = {f"{product}_{id}": id for product, id in zip(products, ids)}
```

Comment:
```suggestion
        filterwheels = {f"{product}_{fw_id}": fw_id for product, fw_id in zip(products, ids)}
```

`id` is a python bulit-in so probably best to avoid using it even in situations where it probably won't matter.